src/content/components/IdleSearchField.js

```diff
@@ -51,23 +44,45 @@ class IdleSearchField extends Component {
     }
   }
 
+  _onClearButtonClick() {
+    this.setState({ value: '' });
+    this._notifyIfChanged('');
+  }
+
+  _onClearButtonFocus(e) {
+    // prevent the focus on the clear button
+    if (e.relatedTarget) {
+      e.relatedTarget.focus();
+    } else {
+      e.target.blur();
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.defaultValue !== this.props.defaultValue) {
+      this._notifyIfChanged(nextProps.defaultValue || '');
+      this.setState({
+        value: nextProps.defaultValue || '',
+      });
+    }
+  }
+
   render() {
-    const { className, title, defaultValue } = this.props;
+    const { className, title } = this.props;
     return (
-      <span className={classNames('idleSearchField', className)}>
-        <input type='search'
+      <form className={classNames('idleSearchField', className)} onSubmit={e => e.preventDefault()}>
+        <input type='search' name='search'
                className='idleSearchFieldInput'
                required='required'
                title={title}
-               defaultValue={defaultValue}
-               ref={this._searchFieldCreated}
-               onInput={this._onSearchFieldInput}
-               onClick={this._onSearchFieldClick}/>
+               value={this.state.value}
+               onChange={this._onSearchFieldChange}
+               onFocus={this._onSearchFieldFocus}/>
         <input type='button'
                className='idleSearchFieldButton'
-               value='Clear'
```

Comment:
good question; initially I changed `type=button` to `type=reset` to try to get the reset behavior for free -- but it doesn't work because this resets the value to what is in `value`... which is being changed by react :)
But I could still use a `type=reset` and let the browser do its job regarding accessibility. What do you think ?